### ISSUE TITLE
(cStor-sparse) App stuck in containerCreating state on node recovery after multi-node failure (ephemeral disk-based pool registration)

### ISSUE THREAD

**Issue by nsathyaseelan** [openebs/openebs#2271]
## Description
Application is seen to remain in containerCreating state due to inconsistencies in the cstor volume target, which was caused as a result of multiple failures on the cluster. 

## Expected Behavior
System/Cluster recovers from failures (such as service crashes, node reboots) etc., and application pod can be rescheduled with data written prior to failures still available.

## Steps to Reproduce
1. Deployed percona application on cstor-sparse pools (volume created with 3 replicas) 
2. Once Application pod entered running status, stopped the node's (say, node-1) kubelet service 
3. This resulted in app pod getting rescheduled on node-2, while a stale pod continued to remain (in `unknown` status) on node-1. The pod was seen to remain in `containerCreating` state on node-2 post the reschedule.
4. Recovered node-1 (kubelet service brought online). With this, the pod entry in `unknown` status also disappeared. 
5. Performed a graceful shutdown of the node-2. Application was node rescheduled on node-1 & remained in `containerCreating` state.
5. Recovered node-2
6. App was still not seen to run. Debug at this point revealed creation of new pools.

Logs for the above
[after_chaos.txt](https://github.com/openebs/openebs/files/2575711/after_chaos.txt)
```
sathyaseelan_n@cloudshell:~ (strong-eon-153112)$ kubectl describe pod percona-786c6b7c7f-rnfz8 -n app-percona-ns
Name:           percona-786c6b7c7f-rnfz8
Namespace:      app-percona-ns
Node:           gke-vitta-cluster-default-pool-93336a3d-4n68/10.128.0.46
Start Time:     Tue, 13 Nov 2018 13:17:58 +0530
Labels:         name=percona
                pod-template-hash=3427263739
Annotations:    <none>
Status:         Pending
IP:
Controlled By:  ReplicaSet/percona-786c6b7c7f
Containers:
  percona:
    Container ID:
    Image:         percona
    Image ID:
    Port:          3306/TCP
    Host Port:     0/TCP
    Args:
      --ignore-db-dir
      lost+found
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:  500m
    Requests:
      cpu:  500m
    Environment:
      MYSQL_ROOT_PASSWORD:  k8sDem0
    Mounts:
      /var/lib/mysql from data-vol (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-r5r8q (ro)
Conditions:
  Type           Status
  Initialized    True
  Ready          False
  PodScheduled   True
Volumes:
  data-vol:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  percona-mysql-claim
    ReadOnly:   false
  default-token-r5r8q:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-r5r8q
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason       Age                From                                                   Message
  ----     ------       ----               ----                                                   -------
  Warning  FailedMount  22m (x24 over 1h)  kubelet, gke-vitta-cluster-default-pool-93336a3d-4n68  MountVolume.WaitForAttach failed for volume "pvc-9131629d-e707-11e8-84fe-42010a800098" : failed to get any path for iscsi disk, last err seen:
iscsi: failed to attach disk: Error: iscsiadm: default: 1 session requested, but 1 already present.
iscsiadm: Could not log into all portals
 (exit status 15)
  Warning  FailedMount  3m (x40 over 1h)  kubelet, gke-vitta-cluster-default-pool-93336a3d-4n68  Unable to mount volumes for pod "percona-786c6b7c7f-rnfz8_app-percona-ns(7079fea3-e718-11e8-84fe-42010a800098)": timeout expired waiting for volumes to attach or mount for pod "app-percona-ns"/"percona-786c6b7c7f-rnfz8". list of unmounted volumes=[data-vol]. list of unattached volumes=[data-vol default-token-r5r8q]
```

## Your Environment

Cluster type: GKE cluster
OS
OpenEBS version: 0.7.1

* `kubectl get nodes`:
```
sathyaseelan_n@cloudshell:~ (strong-eon-153112)$ kubectl get nodes
NAME                                           STATUS    ROLES     AGE       VERSION
gke-vitta-cluster-default-pool-93336a3d-4n68   Ready     <none>    2h        v1.10.6-gke.2
gke-vitta-cluster-default-pool-93336a3d-pqkr   Ready     <none>    2h        v1.10.6-gke.2
gke-vitta-cluster-default-pool-93336a3d-xt01   Ready     <none>    4h        v1.10.6-gke.2
```

* `kubectl get pods --all-namespaces`:
```
sathyaseelan_n@cloudshell:~ (strong-eon-153112)$ kubectl get pods --all-namespaces
NAMESPACE        NAME                                                              READY     STATUS              RESTARTS   AGE
app-percona-ns   percona-786c6b7c7f-rnfz8                                          0/1       ContainerCreating   0          2h
kube-system      event-exporter-v0.2.1-5f5b89fcc8-tz2v9                            2/2       Running             0          2h
kube-system      fluentd-gcp-scaler-7c5db745fc-2kb76                               1/1       Running             0          4h
kube-system      fluentd-gcp-v3.1.0-b2txk                                          2/2       Running             0          2h
kube-system      fluentd-gcp-v3.1.0-k2b78                                          2/2       Running             0          4h
kube-system      fluentd-gcp-v3.1.0-wvpcd                                          2/2       Running             0          4h
kube-system      heapster-v1.5.3-574b4fc8cd-2x7qh                                  3/3       Running             0          2h
kube-system      kube-dns-788979dc8f-55flt                                         4/4       Running             0          4h
kube-system      kube-dns-788979dc8f-6lnbq                                         4/4       Running             0          2h
kube-system      kube-dns-autoscaler-79b4b844b9-qjrp4                              1/1       Running             0          2h
kube-system      kube-proxy-gke-vitta-cluster-default-pool-93336a3d-4n68           1/1       Running             0          2h
kube-system      kube-proxy-gke-vitta-cluster-default-pool-93336a3d-pqkr           1/1       Running             0          1h
kube-system      kube-proxy-gke-vitta-cluster-default-pool-93336a3d-xt01           1/1       Running             0          4h
kube-system      l7-default-backend-5d5b9874d5-zqwtb                               1/1       Running             0          2h
kube-system      metrics-server-v0.2.1-7486f5bd67-bdcl5                            2/2       Running             0          4h
openebs          cstor-sparse-pool-7rhu-c4b6f6c5d-4sjzq                            2/2       Running             0          2h
openebs          cstor-sparse-pool-jpe2-dc7d5597b-sbtvs                            2/2       Running             0          4h
openebs          cstor-sparse-pool-t2qy-696d559894-njffw                           2/2       Running             0          2h
openebs          maya-apiserver-78db8786d7-lfdqr                                   1/1       Running             0          2h
openebs          openebs-ndm-4tbr5                                                 1/1       Running             0          4h
openebs          openebs-ndm-rxrzp                                                 1/1       Running             0          2h
openebs          openebs-ndm-sgrfb                                                 1/1       Running             0          4h
openebs          openebs-provisioner-6f7994f87b-f9chj                              1/1       Running             0          2h
openebs          openebs-snapshot-operator-5c998cddbd-bl7dd                        2/2       Running             0          2h
openebs          pvc-9131629d-e707-11e8-84fe-42010a800098-target-5bb575788dpmgmb   3/3       Running             0          4h
```

**Comment by vishnuitta** [comment]
@BalaBalaYi apologies in responding late to your query. Got to see this message just now.
Error that you mentioned `Error: iscsiadm: default: 1 session requested, but 1 already present.
iscsiadm: Could not log into all portals` is related to iscsi login. This is a timing issue when a login is successful, and IOs are yet to start on the iSCSI disk. In 0.8 openebs, symptom that you mentioned is fixed with PR: https://github.com/openebs/istgt/pull/188. Let me know if you have any other concerns on this.

**Comment by vishnuitta** [comment]
Fixes are available in 0.8.1(211) and master(206). Hence closing.
